app/services/scholars_archive/ordered_parser_service.rb

```diff
@@ -0,0 +1,15 @@
+module ScholarsArchive
+  class OrderedParserService
+    def self.parse(labels)
+      labels ||= []
+      parsed_labels = []
+      sorted_labels = labels.map{ |label| build_array(label)}.sort_by{ |array| array[1] }.map{ |array| array[0] }
+    end
+
+    private
+
+    def build_array(label)
```

Comment:
It looks like this should be 
```
def self.build_array(label)
```
to prevent `undefined method build_array' for ScholarsArchive::OrderedParserService:Class`

Reply:
Yeah, I needed `self.` :P